ospf6d/ospf6_route.c

```diff
@@ -340,19 +329,29 @@ int ospf6_route_get_first_nh_index(struct ospf6_route *route)
 	return (-1);
 }
 
+static int ospf6_nexthop_cmp(struct ospf6_nexthop *a, struct ospf6_nexthop *b)
+{
+	if ((a)->ifindex == (b)->ifindex &&
+	    IN6_ARE_ADDR_EQUAL(&(a)->address, &(b)->address))
+		return 1;
+	return 0;
+}
+
 struct ospf6_route *ospf6_route_create(void)
 {
 	struct ospf6_route *route;
 	route = XCALLOC(MTYPE_OSPF6_ROUTE, sizeof(struct ospf6_route));
 	route->nh_list = list_new();
+	route->nh_list->cmp = (int (*)(void *, void *))ospf6_nexthop_cmp;
+	route->nh_list->del = (void (*) (void *))ospf6_nexthop_delete;
 	return route;
 }
 
 void ospf6_route_delete(struct ospf6_route *route)
 {
 	if (route) {
-		ospf6_free_nexthops(route->nh_list);
-		list_free(route->nh_list);
+		if (route->nh_list)
+			list_free(route->nh_list);
```

Comment:
you need to use `list_delete` here, `list_free` doesn't do shit... (honestly `list_free` shouldn't be in the API to begin with)